### PR TITLE
battery-util: minor follow-ups

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

**Review by DaanDeMeyer — APPROVED** [review]

File: src/battery-check/battery-check.c [hunk 1]
```diff
@@ -153,9 +136,9 @@ static int run(int argc, char *argv[]) {
 
         log_setup();
 
-        r = proc_cmdline_parse(parse_proc_cmdline_item, NULL, PROC_CMDLINE_STRIP_RD_PREFIX);
+        r = proc_cmdline_get_bool("systemd.battery-check", PROC_CMDLINE_STRIP_RD_PREFIX, &arg_doit);
```
Comment (yuwata) [inline src/battery-check/battery-check.c]: Be careful, this disables battery check when the option is not specified. Please re-enable when `r==0`.

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/basic/proc-cmdline.c [hunk 2]
```diff
@@ -402,7 +402,9 @@ int proc_cmdline_get_bool(const char *key, ProcCmdlineFlags flags, bool *ret) {
         if (r < 0)
                 return r;
         if (r == 0) { /* key not specified at all */
-                *ret = false;
+                if (!skip_missing) /* else the original value of ret is retained */
+                        *ret = false;
+
                 return 0;
```
Comment (yuwata) [inline src/basic/proc-cmdline.c]: Hmm, if `skip_missing` is true, then `ret` is not initialized...
How about to rename the flag, say, `missing_to_false` or so, and
```c
if (r == 0) {
        if (!missing_to_false)
                return -ENOKEY;

        *ret = false
        return 0;
}
```

File: src/battery-check/battery-check.c [hunk 3]
```diff
@@ -153,9 +136,9 @@ static int run(int argc, char *argv[]) {
 
         log_setup();
 
-        r = proc_cmdline_parse(parse_proc_cmdline_item, NULL, PROC_CMDLINE_STRIP_RD_PREFIX);
+        r = proc_cmdline_get_bool_full("systemd.battery-check", PROC_CMDLINE_STRIP_RD_PREFIX, /* skip_missing = */ true, &arg_doit);
         if (r < 0)
```
Comment (yuwata) [inline src/battery-check/battery-check.c]: Then, `if (r < 0 && r != -ENOKEY)`.

File: src/shared/fstab-util.c [hunk 4]
```diff
@@ -19,7 +19,7 @@
 
 bool fstab_enabled_full(int enabled) {
         static int cached = -1;
-        bool val;
+        bool val = true;
```
Comment (yuwata) [inline src/shared/fstab-util.c]: Please add a short comment that defaults to true.

File: src/shared/reboot-util.c [hunk 5]
```diff
@@ -110,14 +110,14 @@ int reboot_with_parameter(RebootFlags flags) {
 }
 
 int shall_restore_state(void) {
```
Comment (yuwata) [inline src/shared/reboot-util.c]: Could you install cache for this? rfkill calls this multiple times.

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/basic/proc-cmdline.c (hunk 2 above)
Comment (yuwata) [reply src/basic/proc-cmdline.c]: Moreover, introducing a new flag, say, `PROC_CMDLINE_MISSING_TO_FALSE` is better, than introducing a boolean parameter.

Comment (yuwata) [reply src/basic/proc-cmdline.c]: Or, to preserve most of existing call, conversely `PROC_CMDLINE_MISSING_TO_FATAL` may be better.

File: src/battery-check/battery-check.c (hunk 3 above)
Comment (YHNdnzj) [reply src/battery-check/battery-check.c]: Reconsidering it, the logic seems weird to first specify `PROC_CMDLINE_MISSING_TO_FATAL` and then ignore the error. And the change got way bigger than I originally imagined...

File: src/shared/reboot-util.c (hunk 5 above)
Comment (YHNdnzj) [reply src/shared/reboot-util.c]: -> #28809

**Review by yuwata — APPROVED** [review]

**Comment by poettering** [comment]
lgtm

**Pull request merged**